package/kedro_viz/integrations/kedro/hooks.py

```diff
@@ -0,0 +1,174 @@
+        """Hooks to be invoked after a data catalog is created.
+
+        Args:
+            catalog: The catalog that was created.
+        """
+
+        self.datasets = catalog._data_sets
+
+    @hook_impl
+    def after_dataset_loaded(self, dataset_name: str, data: Any):
+        """Hook to be invoked after a dataset is loaded from the catalog.
+        Once the dataset is loaded, extract the required dataset statistics.
+        The hook currently supports (pd.DataFrame) dataset instances
+
+        Args:
+            dataset_name: name of the dataset that was loaded from the catalog.
+            data: the actual data that was loaded from the catalog.
+        """
+
+        self.create_dataset_stats(dataset_name, data)
+
+    @hook_impl
+    def after_dataset_saved(self, dataset_name: str, data: Any):
+        """Hook to be invoked after a dataset is saved to the catalog.
+        Once the dataset is saved, extract the required dataset statistics.
+        The hook currently supports (pd.DataFrame) dataset instances
+
+        Args:
+            dataset_name: name of the dataset that was saved to the catalog.
+            data: the actual data that was saved to the catalog.
+        """
+
+        self.create_dataset_stats(dataset_name, data)
+
+    @hook_impl
+    def after_pipeline_run(self):
+        """Hook to be invoked after a pipeline runs.
+        Once the pipeline run completes, write the dataset
+        statistics to stats.json file
+
+        """
+        try:
+            with open("stats.json", "w", encoding="utf8") as file:
+                sorted_stats_data = {
+                    dataset_name: self.format_stats(stats)
+                    for dataset_name, stats in self._stats.items()
+                }
+                json.dump(sorted_stats_data, file)
+
+        except Exception as exc:  # pragma: no cover
+            logger.warning(
+                "Unable to write dataset statistics for the pipeline: %s", exc
+            )
+
+    def create_dataset_stats(self, dataset_name: str, data: Any):
+        """Helper method to create dataset statistics.
+        Currently supports (pd.DataFrame) dataset instances.
+
+        Args:
+            dataset_name: The dataset name for which we need the statistics
+            data: Actual data that is loaded/saved to the catalog
+
+        """
+        try:
+            import pandas as pd  # pylint: disable=import-outside-toplevel
+
+            stats_dataset_name = self.get_stats_dataset_name(dataset_name)
+
+            if isinstance(data, pd.DataFrame):
+                self._stats[stats_dataset_name]["rows"] = int(data.shape[0])
+                self._stats[stats_dataset_name]["columns"] = int(data.shape[1])
+
+                current_dataset = self.datasets.get(dataset_name, None)
+
+                if current_dataset:
+                    self._stats[stats_dataset_name]["file_size"] = self.get_file_size(
+                        current_dataset
+                    )
```

Comment:
Good start ! However it only works with dataframes.

There are 2 additional cases that can be easily implemented :
- In the case of a PartitionedDataSet that loads pd.DataFrame s
- In the case of loading multiple sheets of an Excel file (load_args: sheet_name: None )

I think it's not too difficult to account for those cases by adding additional logic on data, and it would avoid just throwing N/A in the UI. 

Do you see how to implement that ?

Reply:
Nice ! thank you